### DISCUSSION TITLE
How to enable paging in RefData?

### DISCUSSION THREAD

**Discussion by karamem0** [Avanade/Beef#183] [Q&A] [answered]
I cannot retrieve fully data in RefData because RefData depends on `BeefDefaultPageSize` parameter. Other entities can be retrieved data using paging, but it seems to not be able to enable paging to RefData. Is it any ideas?

**Comment by chullybun** [comment, marked as answer]
Hi @karamem0, that is by design as it is expected that ref data is cached in its entirety; therefore, paging is not included. What is your scenario as to why paging is needed?

**Reply by karamem0** [reply]
Hi @chullybun,
I am using Cosmos DB template.
For example, I have set `BeefDefaultPageSize` to 20, RefData limited 20 items if Cosmos DB container has more than 20 items. There are not way to retrieve more than 20 items.

**Reply by chullybun** [reply]
hi @karamem0, make sure you are using the latest `Beef.CodeGen.Core` package `v4.2.17` and re-gen code, as that should have been resolved already; see https://github.com/Avanade/Beef/blob/master/tools/Beef.CodeGen.Core/CHANGELOG.md#v4217

**Reply by karamem0** [reply]
Many thanks!
